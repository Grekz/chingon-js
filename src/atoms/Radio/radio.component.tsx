import styled from '@emotion/styled'
import * as React from 'react'

interface IInputProps {
  id: string
  isChecked?: boolean
  type?: string
}

type InputType = React.SFC<IInputProps>

interface ILabelProps {
  children: React.ReactNode
  for?: string
}

interface IRadioProps extends IInputProps, ILabelProps {}

type LabelType = React.SFC<ILabelProps>

const Radio = ({ isChecked = false, children, id }: IRadioProps) => (
  <RadioWrapper>
    <RadioInput id={id} type="radio" isChecked={isChecked} />
    <RadioLabel for={id}>{children}</RadioLabel>
  </RadioWrapper>
)

const RadioWrapper = styled.div`
  position: relative;
  display: block;
`

const RadioInput: InputType = styled.input`
  box-sizing: border-box;
  padding: 0;
`

const RadioLabel: LabelType = styled.label`
  display: inline-block;
`

export default Radio
