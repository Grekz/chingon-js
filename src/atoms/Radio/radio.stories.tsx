import * as React from 'react'

// tslint:disable-next-line: no-implicit-dependencies
import { storiesOf } from '@storybook/react'

import Radio from '.'

storiesOf('Atoms|Radio', module).add('As Default', () => (
  <Radio id="myRadio">Radio button</Radio>
))
