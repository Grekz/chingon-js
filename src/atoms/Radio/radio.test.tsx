import { configure, shallow } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import serializer, { matchers } from 'jest-emotion'
import * as React from 'react'
import Radio from '.'

configure({ adapter: new Adapter() })
expect.extend(matchers)
expect.addSnapshotSerializer(serializer)

describe('<Radio /> rendering', () => {
  it('should render Radio', () => {
    const wrapper = shallow(<Radio id="myRadio">Radio button</Radio>)
    expect(wrapper).toHaveLength(1)
  })
  it('should match snapshot', () => {
    const tree = shallow(<Radio id="myRadio">Radio button</Radio>)
    expect(toJson(tree)).toMatchSnapshot()
  })
})
