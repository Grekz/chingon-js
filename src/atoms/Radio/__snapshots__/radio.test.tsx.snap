// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio /> rendering should match snapshot 1`] = `
<Styled(div)>
  <Styled(input)
    id="myRadio"
    isChecked={false}
    type="radio"
  />
  <Styled(label)
    for="myRadio"
  >
    Radio button
  </Styled(label)>
</Styled(div)>
`;
